### PR TITLE
convert remaining Futures in directory dao to IO

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/CA-329
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAOSpec.scala [hunk 1]
```diff
@@ -85,7 +85,7 @@ class LdapDirectoryDAOSpec extends FlatSpec with Matchers with TestSupport with
       dao.loadUser(user.id).unsafeRunSync()
     }
 
-    runAndWait(dao.deleteUser(user.id))
+    dao.deleteUser(user.id).unsafeRunSync()
```
Comment (helgridly) [inline src/test/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAOSpec.scala:88]: scalatest has a `Futures` mixin that lets you do `someFuture.futureValue`, which is IMO cleaner than `runAndWait`. is there a cats equivalent?

Comment (dvoet) [reply src/test/scala/org/broadinstitute/dsde/workbench/sam/directory/LdapDirectoryDAOSpec.scala:88]: unsafeRunSync _is_ the cats equivalent

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest

**Pull request merged**